# This config file is used by ros2_control
controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    arm_group_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    gripper_group_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    qos_overrides:
      /parameter_events:
        publisher:
          depth: 1000
          durability: volatile
          history: keep_last
          reliability: reliable

      joint_states:
        publisher:
          reliability: best_effort
          durability: volatile
          history: keep_last
          depth: 10

arm_group_controller:
  ros__parameters:
    joints:
      - Joint1
      - Joint2
      - Joint3
      - Joint4
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
gripper_group_controller:
  ros__parameters:
    joints:
      - JointGL
      - JointGR
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity